module/path_helper.go

```diff
@@ -179,41 +179,29 @@ func (m *pathHelperModule) Execute(files map[string]pgs.File, pkgs map[string]pg
 		m.AddGeneratorTemplateFile(dir.Push(dir.Base()).SetExt(".pb.util.fm.go").String(), template.Must(template.New("util").Parse(`package {{ .Package }}
 
 import (
-	"sort"
 	"strings"
 )
 
 // _processPaths returns paths as a pathMap.
 func _processPaths(paths []string) map[string][]string {
-	sort.Strings(paths)
-
-	topLevel := make(map[string]struct{}, len(paths))
-	_pathMap := make(map[string]map[string]struct{}, len(paths))
+	if len(paths) == 0 {
+		return nil
+	}
+	pathMap := make(map[string][]string, len(paths))
 	for _, p := range paths {
 		if !strings.Contains(p, ".") {
```

Comment:
This should be done in a second loop after `pathMap` is populated by nested paths - the result should be independent of path order.

Reply:
Could you show me some case when this function fail to work?

Reply:
I added a test case to `master`: https://github.com/TheThingsIndustries/protoc-gen-fieldmask/commit/65cbe748b16c20eb03b89d5ef96ccace43efdd09
If you rebase your branch on top of latest `master`, you will see that your implementation fails the testcase

Reply:
fixed